cooltools/lib/common.py

```diff
@@ -381,3 +385,631 @@ def print_available_datasets(url=URL_DATA):
+            except Exception as e:
+                raise ValueError(
+                        "expected_df does not match the expected schema:\n"
+                        f"columns {expected_columns} cannot be cast to required data types."
+                    ) from e
+        # raise special message for the old formatted expected_df :
+        elif set(["region","chrom"]).intersection(expected_df.columns):
+            warnings.warn(
+                    "The expected dataframe appears to be in the old format."
+                    "It should have `region1` and `region2` columns instead of `region` or `chrom`."
+                    "Please recalculated your expected using current vestion of cooltools."
+                )
+            raise ValueError(
+                "The expected dataframe appears to be in the old format."
+                "It should have `region1` and `region2` columns instead of `region` or `chrom`."
+                "Please recalculated your expected using current vestion of cooltools."
+            )
+        # does not look like expected at all :
+        else:
+            raise ValueError(
+                "expected_df does not match the expected schema:\n"
+                f"required columns {expected_columns} are missing"
+            )
+
+        # make sure there is no missing data in grouping columns
+        if expected_df[grouping_columns].isna().any().any():
+            raise ValueError(
+                f"There are missing values in columns {grouping_columns}"
+                )
+
+        # make sure region1/2 groups have 1 value for trans contacts
+        # and more than 1 values for cis contacts
+        region1_col, region2_col = grouping_columns[:2]
+        for (r1, r2), df in expected_df.groupby([region1_col, region2_col]):
+            if contact_type == "trans":
+                if len(df) != 1:
+                    ValueError(
+                        f"region {r1},{r2} has more than a single value.\n"
+                        "It has to be single for trans-expected"
+                        )
+                if r1 == r2:
+                    ValueError(
+                        f"region {r1},{r2} is symmetric\n"
+                        "trans expected is caluclated for asymmetric regions only"
+                        )
+            if contact_type == "cis":
+                # generally there shoud be >1 values per region in cis-expected, but
+                # tiny regions smaller than a binsize could have 1 value
+                if len(df) < 1:
+                    ValueError(
+                        f"region {r1},{r2} has to have at least one values for cis-expected"
+                        )
+
+    except Exception as e:
+        if raise_errors:
+            raise e
+        else:
+            # does not look like proper expected
+            return False
+    else:
+        # if no exceptions were raised, it looks like expected_df
+        return True
+
+
```

Comment:
steal the code from:
https://more-itertools.readthedocs.io/en/stable/_modules/more_itertools/more.html#is_sorted